letsencrypt-auto-source/pieces/fetch.py

```diff
@@ -49,14 +50,14 @@ def __init__(self):
             if isinstance(handler, HTTPHandler):
                 self._opener.handlers.remove(handler)
 
-    def get(self, url):
+    def get(self, url, timeout=socket._GLOBAL_DEFAULT_TIMEOUT):
```

Comment:
@cj-dev why not use socket.getdefaulttimeout()?

Reply:
I remember pondering this for a while, but I'm embarrassed I didn't take complete notes on my rationale. A few python issues seem to indicate this is a long-standing debacle [1][2][3] that leaves everyone a little dissatisfied. If I understand correctly, `_GLOBAL_DEFAULT_TIMEOUT` rose to prominence as a sentinel to distinguish between `None` meaning "revert to default" (old) and `None` meaning "no timeout" (present).

The python socket documentation states the default returned by getdefaulttimeout is always going to be `None`, presumably until it is set using setdefaulttimeout [4], which I think would be least useful.

So my justification boiled down to `HttpsGetter` being not much more than a fit-for-purpose `urllib2.urlopen`. I thought that the safest thing to do in my confusion would be to mimic the `urllib2.urlopen` and `OpenerDirector.open` method signatures. When I was looking for precedent to contradict my plan, I didn't see any python in the stdlib that actually uses `getdefaulttimeout`, outside of tests.

If you have a preference or think I missed the mark, just say the word.

1. https://bugs.python.org/issue18417
2. https://bugs.python.org/issue4079
3. https://bugs.python.org/issue14425
4. https://docs.python.org/2/library/socket.html?highlight=socket#socket.getdefaulttimeout

Reply:
Nope! I think you are correct. That is certainly the default value for `timeout` kwarg of `OpenerDirector.open`.  cpython 2.7 source [here](https://github.com/python/cpython/blob/a72d15c97f9837f6c9c6bd476a62175c942cc588/Lib/urllib2.py#L411), master source [here](https://github.com/python/cpython/blob/a72d15c97f9837f6c9c6bd476a62175c942cc588/Lib/urllib2.py#L411).

I was worried this might break some non-cpython pythons, but pypy does it too, and idk if we even need to worry about non-cpython pythons. 👍